crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -211,20 +212,44 @@ impl SolverTimeout {
     /// time left for the driver to forward the results back to the protocol
     /// or do some other necessary work.
     pub fn for_solving(
-        deadline: competition::auction::Deadline,
+        deadline: auction::Deadline,
         now: time::Now,
     ) -> Result<SolverTimeout, auction::DeadlineExceeded> {
-        let deadline = chrono::DateTime::from(deadline) - now.now() - Self::solving_time_buffer();
-        deadline
-            .to_std()
-            .map(Self)
-            .map_err(|_| auction::DeadlineExceeded)
+        Self::with_time_buffer(deadline.into(), now, Self::solving_time_buffer())
+            .ok_or(auction::DeadlineExceeded)
+    }
+
+    /// The time limit passed to the solver when quoting.
+    ///
+    /// Similar to [`SolverTimeout::for_solving`] but for handling quotes. The
+    /// reason for a different value is that less post-processing work is
+    /// required for quoting, meaning a smaller "time buffer" is needed on the
+    /// auction deadline.
+    pub fn for_quoting(
```

Comment:
There is a reason why I initially left the quoting deadline calculation away from the `solution`. The quoting process depends on the solving process. On the contrary, the solving process shouldn't know anything about quoting. Quote-related methods do not belong here.

Reply:
Very good point.

Reply:
Moved. c9409ae3fb16acf921bfcb507bf0865add9728ae